From c8ec261c86b8cd5620a0fb2c2713e7e46a3ec7bf Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Wed, 28 Dec 2022 14:10:24 -0800
Subject: [PATCH] Parser PID Parameter initialize
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Initialize with some defaults in case parsing of Kp, Kd and Ki fails.
---
 src/thd_parse.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/thd_parse.cpp b/src/thd_parse.cpp
index 3042bc3..bf31cd3 100644
--- a/src/thd_parse.cpp
+++ b/src/thd_parse.cpp
@@ -178,6 +178,7 @@ int cthd_parse::parse_new_trip_cdev(xmlNode * a_node, xmlDoc *doc,
 				} else if(!strcasecmp((const char*) cur_node->name,
 						"PidControl")) {
 					pid_control_t pid_params;
+
 					parse_pid_values(cur_node->children, doc, &pid_params);
 					trip_cdev->pid_param.kp = pid_params.Kp;
 					trip_cdev->pid_param.ki = pid_params.Ki;
@@ -293,6 +294,10 @@ int cthd_parse::parse_pid_values(xmlNode * a_node, xmlDoc *doc,
 	xmlNode *cur_node = NULL;
 	char *tmp_value;
 
+	pid_ptr->Kp = 0.0005;
+	pid_ptr->Ki = 0.0001;
+	pid_ptr->Kd = 0.0001;
+
 	for (cur_node = a_node; cur_node; cur_node = cur_node->next) {
 		if (cur_node->type == XML_ELEMENT_NODE) {
 			DEBUG_PARSER_PRINT("node type: Element, name: %s value: %s\n", cur_node->name, xmlNodeListGetString(doc, cur_node->xmlChildrenNode, 1));
-- 
2.30.2

